pexpect/serial_spawn.py

```diff
@@ -0,0 +1,99 @@
+import os
+import sys
+import time
+import threading
+from pexpect.fdpexpect import SpawnBase
+from pexpect.exceptions import EOF, TIMEOUT, ExceptionPexpect
+
+PY3 = (sys.version_info[0] >= 3)
+
+
+class SerialSpawn(SpawnBase):
+    '''This is a serial interface class interface for Pexpect. Use this class to
+    control an existing python serial connection. '''
+    def __init__(self, ser, args=None, **kwargs):
+        if not ser.is_open:
+            raise ExceptionPexpect('serial port is not ready')
+        super().__init__(None, **kwargs)
+        self.ser = ser
+        self.name = f'<serial port {ser.port}>'
+        self.closed = False
+        self.interacting = threading.Event()
+
+    def close(self):
+        if not self.ser.is_open:
+            return
+        self.flush()
+        self.ser.close()
+        self.closed = True
+
+    def isalive(self):
+        return self.ser.is_open
+
+    def send(self, s):
+        s = self._coerce_send_string(s)
+        self._log(s, 'send')
+        b = self._encoder.encode(s, final=False)
+        return self.ser.write(b)
+    
+    def read_nonblocking(self, size=1, timeout=-1):
+        if not self.isalive():
+            raise EOF('End Of File (EOF).')
+        if not timeout:
+            timeout = -1
+        limit_time = time.time() + timeout
+        while self.ser.in_waiting <= 0  and time.time() < limit_time:
```

Comment:
why not set serial.timeout before serial.read()?

Reply:
Nice suggestion. Implemented.